ui/components/MesheryPlayComponent.js

```diff
@@ -71,13 +71,20 @@ class MesheryPlayComponent extends React.Component {
   }
 
   static getDerivedStateFromProps(props, state) {
-    let { meshAdapters, meshAdaptersts, k8sconfig } = props;
+    let { meshAdapters, meshAdaptersts, k8sconfig, selectedAdapter } = props;
+    let adapter  = state.adapter;
+    let aName = selectedAdapter.toLowerCase()
+    if (meshAdapters && meshAdapters.length > 0) {
+      adapter = meshAdapters.filter((meshadapters) => {
+        return aName ===  meshadapters.name.toLowerCase()
+      });
+    }
     const st = {};
     if (meshAdaptersts > state.mts) {
       st.meshAdapters = meshAdapters;
       st.mts = meshAdaptersts;
       if (meshAdapters && meshAdapters.length > 0) {
-        st.adapter = meshAdapters[0];
+        st.adapter = adapter[0];
```

Comment:
Because of explicitly setting the first adapter in the available adapter list, clicking on navigation doesn't change the lifecycle page.

Reply:
Good work. This has been a pain for some time.